framework/src/ble_manager/ble_manager_api.c

```diff
@@ -0,0 +1,338 @@
+/****************************************************************************
+ *
+ * Copyright 2021 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <stdio.h>
+#include <stdint.h>
+#include <ble_manager/ble_manager.h>
+#include "ble_manager_event.h"
+#include "ble_manager_msghandler.h"
+
+#define RETURN_RESULT(res, msg)      \
+	do {                             \
+		if (res < 0) {               \
+			return BLE_MANAGER_FAIL; \
+		} else {                     \
+			return msg.result;       \
+		}                            \
+	} while (0)
+
+ble_result_e ble_manager_init(ble_client_init_config *client_config, ble_server_init_config *server_config)
```

Comment:
As I wrote before,  null check for parameter must be here, because this is api layer , it could be called by application.
Please consider other apis too.